R/choices_labeled.R

```diff
@@ -171,35 +175,48 @@ choices_labeled <- function(choices, labels, subset = NULL, types = NULL) {
 #' })
 #' @export
 #'
-variable_choices <- function(data, subset = NULL, fill = FALSE, key = NULL) {
+variable_choices <- function(data = "all", subset = function(data) names(data), fill = FALSE, key = NULL) {
   checkmate::assert(
     checkmate::check_character(subset, null.ok = TRUE, any.missing = FALSE),
-    checkmate::check_function(subset)
+    checkmate::check_function(subset, args = "data")
```

Comment:
Since function is now default, why allow `NULL`?
```suggestion
    checkmate::check_character(subset, any.missing = FALSE),
    checkmate::check_function(subset, args = "data")
```
Also remove `NULL` provisions in `delayed_choices`.